src/app/components/dynamicdialog/dynamicdialog.ts

```diff
@@ -18,44 +18,113 @@ const hideAnimation = animation([
 ]);
 
 @Component({
-	selector: 'p-dynamicDialog',
-	template: `
-        <div #mask [ngClass]="{'p-dialog-mask':true, 
-            'p-component-overlay p-component-overlay-enter p-dialog-mask-scrollblocker': config.modal !== false, 
-            'p-dialog-left': position === 'left', 
+    selector: 'p-dialog-title',
+    template:`<div class="p-dialog-header" (mousedown)="initDrag($event)">
+    <span class="p-dialog-title">{{config.header}}</span>
+    <div class="p-dialog-header-icons">
+      <button *ngIf="config.maximizable" type="button" [ngClass]="{'p-dialog-header-icon p-dialog-header-maximize p-link':true}" (click)="maximize()" (keydown.enter)="maximize()" tabindex="-1" pRipple>
+        <span class="p-dialog-header-maximize-icon" [ngClass]="maximized ? minimizeIcon : maximizeIcon"></span>
+      </button>
+      <button [ngClass]="'p-dialog-header-icon p-dialog-header-maximize p-link'" type="button" (click)="hide()" (keydown.enter)="hide()" *ngIf="config.closable !== false">
```

Comment:
I think ngClass is not needed here, you have static classes. (The ng class has a doCheck hook, which is not good for performance.)

```html
<button class="p-dialog-header-icon p-dialog-header-maximize p-link" type="button" 
```

Reply:
This was copied as-is from the original but I've changed it now regardless.